text/0087-graphql-errors.md

```diff
@@ -0,0 +1,257 @@
+
+This feature is opt-in by default due to PII concerns.
+
+# Motivation
+
+The [Request](https://develop.sentry.dev/sdk/event-payloads/types/#request) interface contains information on a HTTP request related to the event.
+
+The `Request` contains the `data` field which is the request body. Can be given as string or structural data of any format.
+
+The [Response](https://develop.sentry.dev/sdk/event-payloads/types/#responsecontext) interface contains information on a HTTP response related to the event.
+
+However, the `Response` interface does not contain the `data` field which is the response body.
+
+# Background
+
+This is necessary to implement the [GraphQL Client Errors](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568) with syntax highlight.
+
+The `Request` and `Response` bodies could contain PII.
+
+Request body example (query language only):
+
+```graphql
+query { 
+  viewer { 
+    login
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "data": {
+    "viewer": {
+      "login": "foo"
+    }
+  }
+}
+```
+
+A Request example with an error (query language only):
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    logi
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "errors": [
+    {
+      "path": [
+        "query",
+        "viewer",
+        "logi"
+      ],
+      "extensions": {
+        "code": "undefinedField",
+        "typeName": "User",
+        "fieldName": "logi"
+      },
+      "locations": [
+        {
+          "line": 4,
+          "column": 5
+        }
+      ],
+      "message": "Field 'logi' doesn't exist on type 'User'"
+    }
+  ]
+}
+```
+
+Using the `locations` and the request body (query language only), we can highlight the error in the request body.
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    -> logi
+  }
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+The Request body can also contain `variables`.
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {
+    "login": "foo"
+  }
+}
+```
+
+Because of that, the `Request` and `Response` bodies should be sent to Sentry.
+
+# Supporting Data
+
+See [JIRA issues](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568?pvs=4#fd0b946340d947eb9bbaa320846ccd6a), not disclosing here because they could contain PII.
+
+Related issues and discussions:
+
+[GraphQL Support](https://github.com/getsentry/sentry/discussions/38913)
+
+[Support for GraphQL errors](https://github.com/getsentry/sentry/issues/33723)
+
+[Send request body for http client integrations and similar](https://github.com/getsentry/team-mobile/issues/41)
+
+# Proposal (Option 1)
+
+The proposal is adding a `data` field in the [Response](https://develop.sentry.dev/sdk/event-payloads/types/#responsecontext) interface.
```

Comment:
Dart seems to use the key `body` instead of `data`: https://github.com/getsentry/sentry-dart/blob/7011abe27ac69bd160bdc6ecf3314974b8340b97/dart/lib/src/protocol/sentry_response.dart#L59

I think `data` is better because it is consistent with `Request`, but I wanted to point out the inconsistency.

Reply:
Good point, I can deprecate `body` and add `data`, both will serialize as `data` anyway.